### PR TITLE
Ingestion refactor and new ingestion_type simulated_bundle (C4-506)

### PR DESCRIPTION
**Pull request by netsettler**
This refactors the some of processors.py so that the boilerplate stuff is now primarily done as methods of SubmissionFolio. The result is that it's easier to grab and reuse standard bits of functionality from one ingestion_type to another.

An ingestion_type of simulated_bundle is created to illustrate this. For example, in the `SubmitCGAP` repo there is a file `submit_cgap/tests/data/simulated_bundle.json` (that is mirrored in `cgap-portal` for reference as `src/encoded/tests/data/documents/simulated_bundle.json`). It contains:
```
{
    "validation_output": ["Some validation stuff"],
    "post_output": ["Some post stuff", "More post stuff"],
    "upload_info": {},
    "result": {"answer": 42},
    "institution": "/institutions/hms-dbmi/",
    "project": "/projects/hms-dbmi/"
}
```
This allows one to do (from the `SubmitCGAP repo`):
```
submit-metadata-bundle submit_cgap/tests/data/simulated_bundle.json -v -s http://localhost:8000 --project /projects/hms-dbmi/ --ingestion_type simulated_bundle
```
(for validation, or without the `-v` for a simulated run). The operation does nothing but returns values specified in the given json data file. The point isn't that, but really to say that this is how you structure a processor. So something that wants to provide data in other formats than a standard metadata-bundle can do so.

Ask me (@netsettler) if this is not clear.

### CODE REVIEW

**Review by willronchetti — COMMENTED** [review]
LGTM, but there is a question here that I think merits an answer prior to merge. Perhaps I am out of the loop on the specification of this detail.

File: src/encoded/ingestion/processors.py [hunk 1]
```diff
@@ -63,51 +121,82 @@ def handle_metadata_bundle(submission: SubmissionFolio):
...
+        validation_output = data.get("validation_output", [])
+        for key, value in [("project", project), ("institution", institution)]:
+            if data.get(key) == value:
+                validation_output.append("The %s is OK" % key)
+            else:
+
+                validation_output.append("Expected %s %s." % (key, value))
+                validated = False
+
+        return validated, validation_output
```
[the remark this answers is not in the document]
Comment (sbreiff) [reply src/encoded/ingestion/processors.py:180]: Kent might have a better answer but I think this is just for a simulated processor type, I think in general validation isn't handled in the processor but in what's called by the processor and may be different on a case by case basis. For metadata bundle submission all the validation code is in src/encoded/submit.py.

[the remark this answers is not in the document]
Comment (willronchetti) [reply src/encoded/ingestion/processors.py:180]: Thanks for clarifying! I'm all good with this then.

[the remark this answers is not in the document]
Comment (netsettler) [reply src/encoded/ingestion/processors.py:180]: This is a sample to show where validation goes in the dataflow. We might want to specify a theory of what validation should accomplish, but informally I think validation should be sufficient checking to make sure that if you actually did the posting it is expected to be successful. In this sample, there is no processing, so I wanted to invent some kind of trivial validation just so i could illustrate that the data and control flow might be different depending on whether you passed validation.

**Review by sbreiff — APPROVED** [review]

**Pull request merged**